tsconfig.json

```diff
@@ -7,13 +7,23 @@
     "sourceMap": true,
     "strict": true,
     "allowJs": true,
-    "jsx": "preserve",
+    "jsx": "react-jsx",
     "esModuleInterop": true,
     "allowSyntheticDefaultImports": true,
     "useDefineForClassFields": true, // required for mobx6 - https://mobx.js.org/installation.html#use-spec-compliant-transpilation-for-class-properties
     "outDir": "ts-out",
     "resolveJsonModule": true,
-    "typeRoots": ["./lib/ThirdParty/**/*.d.ts"],
+    "typeRoots": [
+      "./lib/ThirdParty/",
+      "./node_modules/",
+      "./node_modules/@types/",
+      "../node_modules/",
+      "../node_modules/@types/",
+      "../../node_modules/",
+      "../../node_modules/@types/",
+      "../../../node_modules/",
+      "../../../node_modules/@types/"
```

Comment:
Curious why need need 3 levels of parent here? Also I believe `./node_modules` makes `./node_modules/@types/` redundant.

Reply:
I included 3 levels because I have an odd workflow using [git worktree](https://git-scm.com/docs/git-worktree) and yarn packages. I end up with many terriajs folders inside 1 TerriaMap at `packages/tjs/<branch-name>` (for WIP, reviews, testing), and then control which one to link, re-run `yarn install` and build. I could use `packages/<branch-name>` but I like to have all my terriajs folders in their own namespace of sorts. I'm trying to think of a disadvantage of having this config compared to running TS4 which did it's own walking

I tried with only `node_modules/` entries and only `node_modules/@types/` entries and neither is sufficient (types for `react` require `node_modules/@types/` and types for `terriajs-cesium` require `node_modules/`).

The reason this change to typeRoots is needed at all is because of a change in TS 5.1 https://devblogs.microsoft.com/typescript/announcing-typescript-5-1/#explicit-typeroots-disables-upward-walks-for-node_modules-types

Reply:
Thanks, I didn't realise scanning upwards was a behaviour pre 5.1.